File: blaze_flutter_sdk/android/src/main/kotlin/com/blaze/flutterblazesdk/players/shared/SharedPlayerTypes.kt
package com.blaze.flutterblazesdk.players.shared

import androidx.annotation.Keep
import com.blaze.flutterblazesdk.utils.parsing.BlazeReactColor
import com.blaze.flutterblazesdk.utils.parsing.BlazeReactImage
import com.blaze.flutterblazesdk.utils.parsing.BlazeReactTitleFont

@Keep
data class BlazeReactFirstTimeSlideCTAStyle(
    val title: String?,
    val backgroundColor: BlazeReactColor?,
    val textColor: BlazeReactColor?,
    val font: BlazeReactTitleFont?,
    val textSize: Float?,
    val cornerRadius: Int?
)

@Keep
data class BlazeReactPlayerButtonCustomImageStates(
    val selectedImage: BlazeReactImage?,
    val unselectedImage: BlazeReactImage?
)

@Keep
data class BlazeReactCaptionsStyle(
    val font: BlazeReactTitleFont?,
    val textSize: Float?,
    val positioning: BlazeReactCaptionsPositioning?
)

@Keep
data class BlazeReactCaptionsPositioning(
    val xPosition: Map<String, Any?>?,
    val yPosition: Map<String, Any?>?
)

@Keep
data class BlazeReactFirstTimeSlideInstructionStyle(
    val headerText: BlazeReactFirstTimeSlideTextStyle?,
    val descriptionText: BlazeReactFirstTimeSlideTextStyle?
)

@Keep
data class BlazeReactFirstTimeSlideTextStyle(
    val text: String?,
    val font: BlazeReactTitleFont?,
    val textSize: Float?,
    val textColor: BlazeReactColor?
)

@Keep
data class BlazeReactPlayerButtonStyle(
    val width: Int?,
    val height: Int?,
    val color: String?,
    val isVisible: Boolean?,
    val scaleType: String?,
    val isVisibleForAds: Boolean?,
    val customImage: BlazeReactPlayerButtonCustomImageStates?
)

@Keep
data class BlazeReactSeekBarStyle(
    val isVisible: Boolean?,
    val backgroundColor: String?,
    val progressColor: String?,
    val height: Int?,
    val cornerRadius: Float?,
    val thumbColor: String?,
    val thumbImage: BlazeReactImage?,
    val thumbSize: Int?,
    val isThumbVisible: Boolean?
)

@Keep
data class BlazeReactCustomActionButtonParams(
    val id: String?,
    val name: String?,
    val appMetadata: Map<String, Any>?,
    val visibilityCondition: Map<String, Any?>?
)

@Keep
data class BlazeReactCustomActionButton(
    val customParams: BlazeReactCustomActionButtonParams?,
    val style: BlazeReactPlayerButtonStyle?
)

File: blaze_flutter_sdk/ios/Sources/Players/Shared/SharedPlayerTypes.swift
//
//  SharedPlayerTypes.swift
//  blaze_flutter_sdk
//
//

import Foundation

// =================================
// SHARED STRUCTS
// =================================

struct BlazeReactFirstTimeSlideCTAStyle: Codable {
    var title: String?
    var backgroundColor: BlazeReactColor?
    var textColor: BlazeReactColor?
    var font: BlazeReactTitleFont?
    var textSize: CGFloat?
    var cornerRadius: CGFloat?
}

struct BlazeReactFirstTimeSlideInstructionStyle: Codable {
    var headerText: BlazeReactFirstTimeSlideTextStyle?
    var descriptionText: BlazeReactFirstTimeSlideTextStyle?
}

struct BlazeReactFirstTimeSlideTextStyle: Codable {
    var text: String?
    var font: BlazeReactTitleFont?
    var textSize: CGFloat?
    var textColor: BlazeReactColor?
}

struct BlazeReactPlayerButtonStyle: Codable {
    var width: CGFloat?
    var height: CGFloat?
    var color: String?
    var isVisible: Bool?
    var isVisibleForAds: Bool?
    var customImage: BlazeReactPlayerButtonCustomImageStates?
}

struct BlazeReactPlayerButtonCustomImageStates: Codable {
    var selectedImage: BlazeReactImage?
    var unselectedImage: BlazeReactImage?
}

struct BlazeReactSeekBarStyle: Codable {
    var isVisible: Bool?
    var backgroundColor: String?
    var progressColor: String?
    var height: CGFloat?
    var cornerRadius: CGFloat?
    var thumbColor: String?
    var thumbImage: BlazeReactImage?
    var thumbSize: CGFloat?
    var isThumbVisible: Bool?
}

// =================================
// SHARED ENUMS
// =================================

enum BlazeReactBottomComponentsAlignment: String, Codable {
    case relativeToContainer = "RelativeToContainer"
    case relativeToPlayer = "RelativeToPlayer"
    case fitCtaBelowPlayer = "FitCtaBelowPlayer"
}

enum BlazeReactPlayerDisplayMode: String, Codable {
    case fixedRatio9x16 = "FixedRatio_9_16"
    case resizeAspectFillCenterCrop = "ResizeAspectFillCenterCrop"
}
